### ISSUE TITLE
Request to clarify relationship of guidelines to web content-based publications

### ISSUE THREAD

**Issue by mattgarrish** [w3c/wcag21#17]
Another significant item for digital publications that came out of the discussion on October 25th is how to understand and apply WCAG in relation to a publication that is comprised of multiple documents (e.g., where you have one "web page" per chapter).
 
For the EPUB accessibility specification, a lot of our effort went into [explaining an EPUB publication](http://www.idpf.org/epub/a11y/#sec-wcag-eval-page-pub) as the equivalent of a set of web pages, with [clarifying techniques](http://www.idpf.org/epub/a11y/techniques/#sec-wcag) that explain potential differences when a single document entity is broken across multiple pages.

EPUB is often compared to a "web site in a box", but despite that being a fair representation of what you find inside the zip container, users see the publication as though it were a single paginated document. Whether a digital publication is obtainable from a single URI by HTTP is complicated by this duality, as it assumes an online world.

With packaged web content, like EPUB, the type of reading system will also affect the state (e.g., a web-based system would have publications accessible at a single URI, but dedicated applications are working off the file system, as an EPUB file doesn't have to live on the web or travel by HTTP to get to users). Moreover, reading systems will often explode the contents of the zip archive when presenting the content, but some extract resources straight from the zip as needed (i.e., the state of an EPUB publication is variable between a single archive and a group of resources at consumption time).

It might be the case that the description of a single URI accessed by HTTP becomes more true in the future, especially with the work going on around web publications, but even in that work there will be packaged offline publications that have a canonical URI for identification but don't live at it.

We skirted this issue for EPUB and only drew a parallel in applicability of the guidelines and practices to what also amounts to web content. It would be helpful if publications more neatly fell into the picture of WCAG applicability, though.

We've been thinking that adding a description of a web publication to the web page definition examples would help, but are unclear about how to approach the offline/packaged case.

**Comment by jasonjgw** [comment]
If I place my digital publication on a Web server as an archived package, e.g., https://example.org/publication.epub then it appears to satisfy the current definition of "Web page", which reads in part:

> a non-embedded resource obtained from a single URI using HTTP

and the remainder of the definition adds any external resources that would be retrieved in the rendering process.

However, if I extract the archive on the server and make its contents available, then suddenly it's a "set of Web pages", unless you think all of the resources are meant to be rendered together, in which case it's still a single "Web page".

If my publication isn't available over HTTP, then it fails to meet the definition. I can't think of a good reason why the definition shouldn't allow for the "file:" URI scheme, for example.

This problem with the definition is not specific to digital publishing or even to the use of compressed archive files on Web servers. Mobile applications built on Web technologies raise similar issues. I wasn't involved in the WCAG2ICT work, but I know these issues have been explored there, and the Working Group should consider what improvements can be made to this definition in subsequent editions of the Guidelines.

**Comment by joshueoconnor** [comment]
Surveyed: https://www.w3.org/2002/09/wbs/35422/ACT_DPUB_Review_Nov_2016/

**Comment by jasonjgw** [comment]
Should we assert that archives or containers that comprise multiple resources should be treated as though they were replaced by their contents (i.e., as though the contents were available as individual resources) for the purpose of applying the definitions of "Web page" and "Set of Web pages"?

This would at least solve the problem in cases in which the archive file is available via HTTP; it wouldn't solve other problems discussed on the list (e.g., an archive file available on the local file system, such as a publication or a mobile application based on Web technologies that has already been downloaded).

**Comment by mattgarrish** [comment]
That's interesting. Conceptually, it's what we're doing by ignoring the packaging of an EPUB.

I've also been wondering if the set of pages definition would be better stated as "created or assembled by the same author, group or organization". That would give more freedom for open textbooks, as I expect there will inevitably be a question of whether the author is only responsible for ensuring the content they authored is accessible and whether text obtained from open sources is exempt.

Some of this hinges on how DPUB addresses compiling a publication from disparate resources across the web, since in EPUB the author has to physically move the content within the container (but still leaves open the question of responsibility for correcting open content). It could wait for silver, in other words, especially if it causes issues for backwards compatibility.

**Comment by joshueoconnor** [comment]
Related to #313

**Comment by awkawk** [comment]
core of issues is in Issue 110 so closing this one.
